HealthcareSystem/Backend/Repository/SurveyRepository/MySqlSurveyRepository/MySqlSurveyRepository.cs

```diff
@@ -24,15 +25,22 @@ public void AddSurvey(Survey survey)
         public List<SurveyResult> GetSurveyResultsAboutDoctor(string jmbg)
         {
             List<SurveyResult> surveyResults = new List<SurveyResult>();
-            SurveyResult behaviorOfDoctor = getSurveyResultAboutBehaviorOfDoctor(jmbg);          
-            SurveyResult doctorProfessionalism = getSurveyResultAboutDoctorProfessionalism(jmbg);
-            SurveyResult gettingAdviceByDoctor = getSurveyResultAboutGettingAdviceByDoctor(jmbg);
-            SurveyResult availabilityOfDoctor = getSurveyResultAboutAvailabilityOfDoctor(jmbg);
-            surveyResults.Add(behaviorOfDoctor);
-            surveyResults.Add(doctorProfessionalism);
-            surveyResults.Add(gettingAdviceByDoctor);
-            surveyResults.Add(availabilityOfDoctor);
-            return surveyResults;
+            try
+            {
+                SurveyResult behaviorOfDoctor = getSurveyResultAboutBehaviorOfDoctor(jmbg);
+                SurveyResult doctorProfessionalism = getSurveyResultAboutDoctorProfessionalism(jmbg);
+                SurveyResult gettingAdviceByDoctor = getSurveyResultAboutGettingAdviceByDoctor(jmbg);
+                SurveyResult availabilityOfDoctor = getSurveyResultAboutAvailabilityOfDoctor(jmbg);
+                surveyResults.Add(behaviorOfDoctor);
+                surveyResults.Add(doctorProfessionalism);
+                surveyResults.Add(gettingAdviceByDoctor);
+                surveyResults.Add(availabilityOfDoctor);
+                return surveyResults;
```

Comment:
So I'm only noticing this now, but, since you're really just doing aggregation in all these methods - consider converting them into group by queries (Linq has a GroupBy method, and Entity Framework should convert it correctly). Or, if this calculation is kept as is, I would maybe consider moving it to the Service.